### PR TITLE
feat: Using refresh token

### PR DESCRIPTION
**Pull request by 0sinf**
## What does this PR do?

Use refresh token for refresh access token that expired.

## Type of change

- [x] New feature
- [ ] Bug fix
- [ ] Code improvements
- [ ] Breaking change
- [ ] Documentation
- [ ] etc.

## Changes

- combined auth and jwt because I think that need to manage in one module.
- When login, send access token and refresh token in cookie.
- In auth controller, send new things or reject request by refresh token.
- Add guards. (access token guard, refresh token guard)
- Add types for JWT

### CODE REVIEW

File: src/@types/auth.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export interface JwtPayload {
+  id: string;
+  email: string;
+}
+
+export interface JwtPayloadWithRT extends JwtPayload {
+  refreshToken: string;
+}
+
+export interface Tokens {
```
Comment (marshallku) [inline src/@types/auth.d.ts]: I don't think this should be plural.

File: src/auth/strategies/rt.strategy.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  validate(req: Request, payload: JwtPayload): JwtPayloadWithRT {
+    const refreshToken = req
+      .get('authorization')
+      ?.replace('Bearer', '')
+      ?.trim();
+
+    return {
+      ...payload,
+      refreshToken,
```
Comment (marshallku) [inline src/auth/strategies/rt.strategy.ts:25]: It's returning `undefined` if `authorization` is not in request.
Did you mean it?

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import * as argon2 from 'argon2';
+import { JwtPayload, Tokens } from 'src/@types/auth';
+
+@Injectable()
+export class AuthService {
+  constructor(private jwtService: JwtService) {}
+
+  login(payload: JwtPayload): Tokens {
+    const accessToken = this.jwtService.sign(payload, {
+      expiresIn: '2h',
```
Comment (marshallku) [inline src/auth/auth.service.ts]: Think it would be good to talk about the expiration time

File: src/auth/strategies/rt.strategy.ts (hunk 2 above)
Comment (0sinf) [reply src/auth/strategies/rt.strategy.ts:25]: If authorization isn't in request, already return that Invalid token.
It defined constructor in strategy.

Is it good that doesn't use optional chaining when get refresh token?

File: src/auth/auth.service.ts (hunk 3 above)
Comment (0sinf) [reply src/auth/auth.service.ts]: Good!

File: src/@types/auth.d.ts (hunk 1 above)
Comment (0sinf) [reply src/@types/auth.d.ts]: OK, I agress.

File: src/auth/strategies/rt.strategy.ts (hunk 2 above)
Comment (marshallku) [reply src/auth/strategies/rt.strategy.ts:25]: nvm if u meant it

**Review by marshallku — APPROVED** [review]
LGTM!

**Pull request merged**